lib/datapacksjob.js

```diff
@@ -2961,12 +2963,69 @@ DataPacksJob.prototype.downloadPerformanceData = async function (jobInfo) {
                 records.push(record);
             })
             .on("end", async function () {
-                var res = records.map(item => {
-                    return item[field_name];
+               
+                var omni_id =records.map(item => {
+                    return JSON.parse(item[field_name]).OmniScriptId;
                 });
 
-                VlocityUtils.log(`${i} of ${totalEntries.totalSize}`);
+                var unique_omni_id=Array.from([...new Set(omni_id)]);
+ 
+                var unique_omni_id_format="";
+                
+                for(var j=0;j<unique_omni_id.length;j++){
+                    if(j == (unique_omni_id.length-1)){
+                        unique_omni_id_format=unique_omni_id_format+`'${unique_omni_id[j]}'`
+                    }else{
+                        unique_omni_id_format=unique_omni_id_format+ `'${unique_omni_id[j]}',`;
+                    }
+                }
+
+                var additional_data = await vloc.jsForceConnection.query(`SELECT ${vloc.namespace}__PropertySet__c,Name,${vloc.namespace}__OmniScriptId__c FROM ${vloc.namespace}__Element__c WHERE ${vloc.namespace}__OmniScriptId__c IN (${unique_omni_id_format})`);
+            
+                var res =[];
 
+                additional_data.records.map(item => {
+
+                    var name=item['Name']
+                    var omniscript_id=item[vloc.namespace+'__OmniScriptId__c'];
+                    var propertyset=JSON.parse(item[vloc.namespace+'__PropertySet__c']);
+                   
+                    records.map(item => {
+                        var element_name= JSON.parse(item[field_name]).ElementName;
+                        var element_omniScript_id= JSON.parse(item[field_name]).OmniScriptId;
+                        var element_type= JSON.parse(item[field_name]).ElementType;
+                        var element_data=JSON.parse(item[field_name])
```

Comment:
This parse should just be called once and then use the parsed object. Also, please use spaces before and after `=`

Reply:
done, Please find updated code below :- 

                            var element_data = JSON.parse(item[field_name])
                            var element_name = element_data.ElementName;
                            var element_omniScript_id = element_data.OmniScriptId;
                            var element_type = element_data.ElementType;